interface/src/app/tribe_client/genesets.js

```diff
@@ -0,0 +1,97 @@
+angular.module('adage.tribe_client.genesets', [
+  'adage.tribe_client.resource'
+])
+
+// Geneset Search Factory
+// Allows for retreiving for and interacting with search results for genesets.
+.factory('GenesetSearch', ['Genesets', function(Genesets) {
+  var genesets = [];
+  var query = {};
+  var resultCount = null;
+
+  return {
+    getQuery: function() {
+      return query;
+    },
+    getGenesets: function() {
+      return genesets;
+    },
+    clear: function() {
+      query = {};
+      genesets = [];
+    },
+    getResultCount: function() {
+      return resultCount;
+    },
+    query: function(searchParams) {
+      query = searchParams;
+      Genesets.query(query, function(data) {
+        resultCount = data.meta.total_count;
+        genesets = data.objects;
+      });
+    }
+  };
+}])
+
+.directive('genesetSearchForm', ['GenesetSearch', function(GenesetSearch) {
+  return {
+    controller: ['$scope', function($scope) {
+      // Number of results (in this case genesets) to come back
+      // in each response page.
+      $scope.limit = 10;
+    }],
+    replace: true,
+    restrict: 'E',
+    scope: {
+      limit: '=',
+      organism: '='
+    },
+    templateUrl: 'tribe_client/geneset-search-form.tpl.html'
+  };
+}])
+
+
+// Search box for users to enter their search text into
+// and go search Tribe genesets.
+.directive('genesetSearchBar', ['GenesetSearch', function(GenesetSearch) {
+  return {
+    controller: ['$scope', function($scope) {
+      $scope.search = {};
+      $scope.search['organism__scientific_name'] = $scope.organism;
+      $scope.search['limit'] = $scope.limit;
+
+      $scope.searchGenesets = function() {
+        GenesetSearch.query($scope.search);
```

Comment:
You might think this line would make up for that, but you have an isolate scope, which means the `genesetSearchBar` cannot see the `.search` you set in the template (parent scope) because you haven't passed it in.

I believe this is why it's not quite working right. Sorry I missed that in review earlier!